.github/workflows/main.yml

```diff
@@ -73,13 +225,42 @@ jobs:
         uses: aws-actions/amazon-ecr-login@v2
         with:
           registries: ${{ secrets.DEPLOY__AWS_ECR_ACCOUNT }}
+      
+      - name: Build deployment target variables
+        id: vars
+        run: |
+          echo "Setting up deployment variables..."
+          echo "IMAGE_URL=${{ secrets.DEPLOY__AWS_ECR_ACCOUNT }}.dkr.ecr.${{ secrets.DEPLOY__AWS_DEFAULT_REGION }}.amazonaws.com/appbuilder-portal-origin" >> $GITHUB_OUTPUT
+          [[ "${{ github.ref }}" == "refs/heads/master" ]] && echo "DEPLOY_LEVEL=production" >> $GITHUB_OUTPUT || echo "DEPLOY_LEVEL=staging" >> $GITHUB_OUTPUT
+          [[ "${{ github.ref }}" == "refs/heads/master" ]] && echo "ECS_CLUSTER=scriptoria-prd" >> $GITHUB_OUTPUT || echo "ECS_CLUSTER=scriptoria-stg" >> $GITHUB_OUTPUT
 
-      - name: Build nginx
-        run: ./scripts/build-nginx.sh $GITHUB_SHA $GITHUB_REF $GITHUB_EVENT_NAME
+      - name: Build Docker image
+        run: |
+          echo "Building Docker image..."
+          docker build -t scriptoria-portal:${{ github.sha }} .
+          echo "Docker image built successfully."
+          docker tag scriptoria-portal:${{ github.sha }} ${{ steps.vars.outputs.IMAGE_URL }}:${{ steps.vars.outputs.DEPLOY_LEVEL }}
+          docker tag scriptoria-portal:${{ github.sha }} ${{ steps.vars.outputs.IMAGE_URL }}:${{ github.sha }}
+          echo "Docker image tagged as ${{ steps.vars.outputs.DEPLOY_LEVEL }} and ${{ github.sha }}"
 
-      - name: Build API
-        run: ./scripts/build-api.sh $GITHUB_SHA $GITHUB_REF $GITHUB_EVENT_NAME
+      - name: Push Docker image to ECR
+        run: |
+          echo "Pushing Docker image to ECR..."
+          docker push ${{ steps.vars.outputs.IMAGE_URL }}:${{ steps.vars.outputs.DEPLOY_LEVEL }}
+          docker push ${{ steps.vars.outputs.IMAGE_URL }}:${{ github.sha }}
+          echo "Docker image pushed successfully to ${{ steps.vars.outputs.IMAGE_URL }}"
 
-      - name: Deploy
-        if: (github.event_name == 'push' || github.event_name == 'workflow_dispatch' || github.event_name == 'repository_dispatch') && (github.ref == 'refs/heads/develop' || github.ref == 'refs/heads/master')
```

Comment:
It seems like deploy should only happen on push or workflow_display (not pull_request) and only on develop or master.